### PR TITLE
added input and select for rhf

### PR DESCRIPTION
**Pull request by Akshaya-Dhelaria-au7**

### CODE REVIEW

File: react-hook-form-mui/src/forms/Select.js [hunk 1]
```diff
@@ -26,21 +24,23 @@ const Select = ({
...
-						{option.label}
+					<option key={i} value={option.code}>
+						{option.name}
 					</option>
 				))
 				: options.map((option, i) => (
-					<MenuItem key={i} value={option.value}>
-						{option.label}
+					<MenuItem key={i} value={option.code}>
+						{option.name}
```
Comment (hardcodeai) [inline react-hook-form-mui/src/forms/Select.js:38]: Why is it option.name and not label? we would be sending options as a pair of key value pairs of the form of [{value:" " ,label: " "},{value:" ",label:" "}]

File: react-hook-form-mui/src/forms/Select.js [hunk 2]
```diff
@@ -26,21 +24,23 @@ const Select = ({
...
-					<option key={i} value={option.value}>
-						{option.label}
+					<option key={i} value={option.code}>
+						{option.name}
 					</option>
 				))
 				: options.map((option, i) => (
-					<MenuItem key={i} value={option.value}>
-						{option.label}
+					<MenuItem key={i} value={option.code}>
```
Comment (hardcodeai) [inline react-hook-form-mui/src/forms/Select.js:37]: Same thing with code, please change the code to value

File: react-hook-form-mui/src/forms/TextField.js [hunk 3]
```diff
@@ -7,17 +7,14 @@ function TextField({compact, InputProps, InputLabelProps = {}, InputAdornmentPro
...
-		};
-		label = '';
+			...label,
+			...(props.adornment && {startAdornment: (
+				<InputAdornment disablePointerEvents position='start' style={{whiteSpace: 'nowrap', fontSize: '0.8rem', opacity: 0.85}} {...InputAdornmentProps}>
+					{props.adornment}
+				</InputAdornment>
+			)})
+		}
+		// label = '';
```
Comment (hardcodeai) [inline react-hook-form-mui/src/forms/TextField.js:17]: Please remove this commented code, also destructure adornment from props and then use it

File: react-hook-form-mui/src/Input.js [hunk 4]
```diff
@@ -58,72 +42,28 @@ const getExtraProps = function({type, label, required, validate: validateOrig, c
...
-	const extraProps = {...(rhf ? {fast, ...(input ? {component: input} : {})} : {}), compact, components, ...getExtraProps({type, label, required: rest.required, validate, compact, rhf, indian: rest.indian})};
-
+	Field = Field || module({type, rhf});
+	const extraProps = {...(rhf ? {...(input ? {component: input} : {})} : {}), compact, label, components, ...getExtraProps({type, label, required: rest.required, validate, compact, rhf, indian: rest.indian})};
 	return (
 		<ErrorBoundary>
 			<Container {...containerProps}>
 				{Field
-					? <Field {...rest} {...(newType ? {type: newType} : {})} {...extraProps}/>
+					? <Field {...rest} {...type} {...extraProps}/>
```
Comment (hardcodeai) [inline react-hook-form-mui/src/Input.js:66]: Why this format ?

File: react-hook-form-mui/src/Input.js [hunk 5]
```diff
@@ -26,30 +27,13 @@ class ErrorBoundary extends React.Component {
 	}
 }
 
-const getExtraProps = function({type, label, required, validate: validateOrig, compact, rhf, indian}) {
-	if (required && compact && label) label = label.replace(/\*$/, '').trim() + ' *';
+const getExtraProps = function({type, label, required, compact, rhf, validate: validateOrig, indian}) {
```
Comment (hardcodeai) [inline react-hook-form-mui/src/Input.js:30]: please add the regex for cleaning the label back

File: react-hook-form-mui/src/Input.js (hunk 4 above)
Comment (Akshaya-Dhelaria-au7) [reply react-hook-form-mui/src/Input.js:66]: Since the field is expecting string.

**Pull request merged**